pkg/action/upgrade.go

```diff
@@ -64,6 +64,8 @@ type Upgrade struct {
 	SkipCRDs bool
 	// Timeout is the timeout for this operation
 	Timeout time.Duration
+	// HookTimeout is the timeout for any hooks in this operation
+	HookTimeout time.Duration
```

Comment:
Additionally to my note on flag behavioral compatibility, Helm SDK users would get a default value of zero (due to zero initialization) for `HookTimeout` (here and Install/Uninstall/Rollback too).

Reply:
Good explanation.
Thanks.

Reply:
This should be fine since with the CLI the default is now zero and it is only used if it is set, and Helm uses the standard timeout if it is not set.